robottelo/cli/hammer.py

```diff
@@ -0,0 +1,163 @@
+"""Helpers to interact with hammer command line utility."""
+import re
+
+from itertools import izip
+
+
+def parse_csv(output):
+    """Parse CSV output from Hammer CLI and convert it to python dictionary."""
+    result = []
+    # Generate the key names, spaces will be converted to dashes "-"
+    keys = [
+        header.replace(' ', '-').lower() for header in output.pop(0).split(',')
+    ]
+
+    # For each entry, split the values and create a dict mapping each key with
+    # each value
+    for line in output:
+        if len(line) > 0:
+            result.append(dict(izip(keys, line.split(','))))
+
+    return result
+
+
+def parse_help(output):
+    """Parse the help output from a hammer command and return a dictinary
+    mapping the subcommands and options accepted by that command.
+
+    """
+    # Parsing states
+    state = 0
+    subcommands_section_state = 1
+    options_section_state = 2
+
+    contents = {
+        'subcommands': [],
+        'options': [],
+    }
+    option_regex = re.compile(
+        r'^ (-(?P<shortname>\w), )?(--(?P<name>[\w-]+))?( (?P<value>\w+))?\s+'
+        '(?P<help>.*)$'
+    )
+    subcommand_regex = re.compile(
+        r'^ (?P<name>[\w-]+)?\s+(?P<description>.*)$'
+    )
+
+    for line in output:
+        if len(line.strip()) == 0:
+            continue
+        if line.startswith('Subcommands:'):
+            state = subcommands_section_state
+            continue
+        if line.startswith('Options:'):
+            state = options_section_state
+            continue
+
+        if state == subcommands_section_state:
+            match = subcommand_regex.search(line)
+            if match is not None:
```

Comment:
Can we make the following change to the logic of this function?

```python
if match is None:
    continue
if match.group('name') is None:
   …
else:
    …
```

I find logic easier to follow when "error" cases are checked for and dealt with than when "success" cases are checked for and dealt with. It tends to produce less indented code, if nothing else.